os/fs/smartfs/smartfs_smart.c

```diff
@@ -1354,6 +1355,12 @@ static int smartfs_opendir(struct inode *mountpt, const char *relpath, struct fs
 		goto errout_with_semaphore;
 	}
 
+	if (!(entry.flags & SMARTFS_DIRENT_TYPE_DIR)) {
```

Comment:
not good idea, you can do this in fs_opendir to apply this for all fs .
please check struct dirent, you can use d_type